includes/data-events/listeners.php

```diff
@@ -301,3 +302,123 @@ function( $subscription, $status_to, $status_from ) {
 		];
 	}
 );
+
+/**
+ * When a non-donation subscription is activated.
+ * The subscription will be added to the user's list of active subscriptions.
+ */
+Data_Events::register_listener(
+	'woocommerce_subscription_status_active',
+	'product_subscription_active',
+	function( $subscription ) {
+		// We only want to fire this for non-donation products.
+		$product_ids = array_values(
+			array_filter(
+				\Newspack\WooCommerce_Connection::get_products_for_subscription( $subscription->get_id() ),
+				function( $product_id ) {
+					return ! Donations::is_donation_product( $product_id );
+				}
+			)
+		);
+
+		return [
+			'user_id'         => $subscription->get_customer_id(),
+			'email'           => $subscription->get_billing_email(),
+			'subscription_id' => $subscription->get_id(),
+			'product_ids'     => $product_ids,
+			'amount'          => (float) $subscription->get_total(),
+			'currency'        => $subscription->get_currency(),
+			'recurrence'      => $subscription->get_billing_period(),
+		];
+	}
+);
+
+/**
+ * When a non-donation subscription is deactivated.
+ * The subscription will be removed from the user's list of active subscriptions.
+ */
+Data_Events::register_listener(
+	'woocommerce_subscription_status_updated',
+	'product_subscription_inactive',
+	function( $subscription, $status_to, $status_from ) {
+		// We only want to fire this for non-active statuses.
+		if ( 'active' === $status_to ) {
+			return;
+		}
+
+		// We only want to fire this for non-donation products.
```

Comment:
Same as in the `woocommerce_subscription_status_active` listener – shouldn't the function return early if `$product_ids` is empty?

Reply:
Good call, 6b26f574 returns early without firing the data event if there are no non-subscription products associated with the subscription.